docs/src/main/sphinx/admin/fault-tolerant-execution.rst

```diff
@@ -0,0 +1,338 @@
+========================
+Fault-tolerant execution
+========================
+
+By default, if a Trino node lacks the resources to execute on a task or
+otherwise fails during query execution, the query fails and must be run again
+manually. The longer the runtime of a query, the more likely it is to be
+susceptible to such failures.
+
+Fault-tolerant execution is a mechanism in Trino that enables a cluster to
+mitigate query failures by retrying queries or their component tasks in
+the event of failure. With fault-tolerant execution enabled, in-progress query
+data is spooled and can be re-used by another worker in the event of a worker
+outage or other fault during query execution.
+
+.. note::
+
+    Fault tolerance does not apply to user error. For example, a query that
+    fails because its SQL cannot be parsed is not fault-tolerant.
+
+Configuration
+-------------
+
+Fault-tolerant execution is disabled by default. To enable the feature, set the
+``retry-policy`` configuration property to either ``QUERY`` or ``TASK``
+depending on the desired :ref:`retry policy <fte-retry-policy>`.
```

Comment:
Technically we can also enable fault tolerance using `retry_policy` session property. But I am not convinced we want to advertise this possibility in documentation. 
WDYT @arhimondr

Reply:
We should consider adding that session property to the `retry-policy` entry I added in `properties-query-management.rst` for completeness, and just not mention it in this page. Unless there's an active reason to dissuade people from using the session property, in which case it should also be marked as "experimental" or equivalent in the engine.

Reply:
IDK. The session property (I think) is there mostly for ease of testing. So we can easily switch the logic and compare behaviour. I would make session property hidden and not document it. WDYT @arhimondr @linzebing ?

Reply:
I think it's important to have in the properties reference just so people who stumble on the property with `SHOW SESSION` can find an explanation in the docs.

Reply:
Yeah, I agree with @losipiuk . Currently the system is not designed to properly handle queries with `retry_policy=NONE|QUERY` and `retry_policy=TASK` running on the same cluster in parallel. Generally our advice should be to deploy a dedicated cluster with `retry_policy=TASK` enabled. Queries running with `retry_policy=QUERY` can coexist with queries running with `retry_policy=NONE`.

Also it feels like it is worth mentioning the tradeoffs when choosing between `QUERY` and `TASK`. 

- `QUERY` - doesn't spool intermediate data and execution is exactly the same as for `NONE`, but in case of a failure the entire query will be transparently restarted.
- `TASK` - requires spooling of intermediate data. Changes execution strategy. For `TASK` a dedicated cluster must be deployed and all query on a given cluster have to be run with `retry_policy=TASK`

Reply:
Got it, that makes sense why we would want to discourage use of the session property. In that case, is there a way to prevent the `retry_policy` session property from appearing as a result of `SHOW SESSION`?

Reply:
That's a good point. People might still want to enable `retry_policy=QUERY` for certain queries but it seems to be a lesser evil. https://github.com/trinodb/trino/pull/11480

Reply:
Perfect, thanks. Removed the `retry_policy` session property from this PR.